examples/classification/benchmark.py

```diff
@@ -0,0 +1,393 @@
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the benchmark on,
+                        defaults to None where it uses all physical cores
+                        available on the system. For DeepSparse benchmarks,
+                        this value is the number of cores per socket
+  -s NUM_SOCKETS, --num-sockets NUM_SOCKETS
+                        For DeepSparse benchmarks only. The number of physical
+                        cores to run the benchmark on. Defaults to None where
+                        is uses all sockets available on the system
+  -i NUM_ITERATIONS, --num-iterations NUM_ITERATIONS
+                        The number of iterations the benchmark will be run for
+  -w NUM_WARMUP_ITERATIONS, --num-warmup-iterations NUM_WARMUP_ITERATIONS
+                        The number of warmup iterations that will be executed
+                        before the actual benchmarking
+
+"""
+import argparse
+import time
+from tempfile import NamedTemporaryFile
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy
+import onnx
+import onnxruntime
+from tqdm import tqdm
+
+from deepsparse import compile_model
+from deepsparse.benchmark import BenchmarkResults
+from examples.classification.helper import _BatchLoader, _load_random_data, load_data
+from sparseml.onnx.utils import (
+    get_tensor_dim_shape,
+    override_model_batch_size,
+    set_tensor_dim_shape,
+)
+from sparsezoo import Zoo
+
+
+DEEPSPARSE_ENGINE = "deepsparse"
+ORT_ENGINE = "onnxruntime"
+
+
+def _run_model(
+    args, model: Any, batch: Union[numpy.ndarray]
+) -> List[Union[numpy.ndarray]]:
+    # run model according to engine type
+    if args.engine == ORT_ENGINE:
+        outputs = model.run(
+            [out.name for out in model.get_outputs()],  # outputs
+            {model.get_inputs()[0].name: batch},  # inputs dict
+        )
+    else:  # deepsparse
+        outputs = model.run(batch)
+    return outputs
+
+
+def _get_data_loader(config, new_image_shape, total_iterations):
+    """
+    Helper function to get appropriate batch data loader
+    """
+    if not config.data_path:
+        data_loader = _load_random_data(
+            batch_size=config.batch_size,
+            iterations=total_iterations,
+            image_size=new_image_shape,
+        )
+    else:
+        dataset = load_data(config.data_path)
+        data_loader = _BatchLoader(dataset, config.batch_size, total_iterations)
+    return data_loader
+
+
+def fix_onnx_input_shape(
+    model_path: str,
+    image_shape: Optional[Tuple[int]],
+) -> Tuple[str, Optional[NamedTemporaryFile]]:
+    """
+    Creates a new ONNX model from the given path that accepts the given input
+    shape. If the given model already has the given input shape no modifications are
+    made. Uses a tempfile to store the modified model file.
+    :param model_path: file path to ONNX model or SparseZoo stub of the model
```

Comment:
add blank line between main info and params. rst may not compile correctly without it

Reply:
👍